--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -590,11 +590,13 @@
 
 	update_fdt_from_fw(blob, (void *)fw_dtb_pointer);
 
+#if CONFIG_IS_ENABLED(VIDEO_DT_SIMPLEFB)
 	node = fdt_node_offset_by_compatible(blob, -1, "simple-framebuffer");
 	if (node < 0)
 		fdt_simplefb_add_node(blob);
 	else
 		fdt_simplefb_enable_and_mem_rsv(blob);
+#endif
 
 #ifdef CONFIG_EFI_LOADER
 	/* Reserve the spin table */
